pootle/core/views.py

```diff
@@ -359,3 +435,226 @@ def page_not_found(request):
+        url_action_view_all = self.object.get_translate_url(state='all')
+        ctx, cookie_data = self.sidebar_announcements
+        ctx.update(
+            super(PootleBrowseView, self).get_context_data(*args, **kwargs))
+        ctx.update(
+            {'page': 'browse',
+             # 'parent': get_parent(self.object),
+             'stats': jsonify(self.stats),
+             'translation_states': get_translation_states(self.object),
+             'check_categories': get_qualitycheck_schema(self.object),
+             'url_action_continue': url_action_continue,
+             'url_action_fixcritical': url_action_fixcritical,
+             'url_action_review': url_action_review,
+             'url_action_view_all': url_action_view_all,
+             'table': self.table,
+             'is_store': self.is_store,
+             'browser_extends': self.template_extends})
+        return ctx
+
+
+class PootleTranslateView(PootleDetailView):
+    template_name = "editor/main.html"
+
+    @property
+    def ctx_path(self):
+        return self.pootle_path
+
+    @property
+    def vfolder_pk(self):
+        return ""
+
+    @property
+    def display_vfolder_priority(self):
+        if 'virtualfolder' not in settings.INSTALLED_APPS:
+            return False
+        return VirtualFolderTreeItem.objects.filter(
+            pootle_path__startswith=self.object.pootle_path).exists()
+
+    def get_context_data(self, *args, **kwargs):
+        ctx = super(
+            PootleTranslateView, self).get_context_data(*args, **kwargs)
+        ctx.update(
+            {'page': 'translate',
+             'current_vfolder_pk': self.vfolder_pk,
+             'profile': self.request.profile,
+             'ctx_path': self.ctx_path,
+             'display_priority': self.display_vfolder_priority,
+             'check_categories': get_qualitycheck_schema(),
+             'cantranslate': check_permission("translate", self.request),
+             'cansuggest': check_permission("suggest", self.request),
+             'canreview': check_permission("review", self.request),
+             'search_form': make_search_form(request=self.request),
+             'previous_url': get_previous_url(self.request),
+             'POOTLE_MT_BACKENDS': settings.POOTLE_MT_BACKENDS,
+             'AMAGAMA_URL': settings.AMAGAMA_URL,
+             'editor_extends': self.template_extends})
+        return ctx
+
+
+class PootleExportView(PootleDetailView):
+    template_name = 'editor/export_view.html'
+
+    @property
+    def path(self):
+        return self.request.path.replace("export-view/", "")
+
+    def get_context_data(self, *args, **kwargs):
+        ctx = {}
+        filter_name, filter_extra = get_filter_name(self.request.GET)
+        units_qs = Unit.objects.get_for_path(self.object.pootle_path,
+                                             self.request.profile)
+        units_qs = get_step_query(self.request, units_qs)
+        unit_total_count = units_qs.count()
```

Comment:
this is a problem with extra fields in the orm query and is unrelated to this PR